### ISSUE TITLE
Japanese input method works well but the carret position is wrong

### ISSUE THREAD

**Issue by kkk669** [blinksh/blink#447]
There is no problem in Japanese input itself but a small bug. Please could you see the following short video? Thanks.

[![YouTube link](https://img.youtube.com/vi/Esme-rO14NI/0.jpg)](https://youtu.be/Esme-rO14NI)

My environment:
- Blink Shell 10.015
- iOS 11.3 developer beta 3 (15E5178f)
- iPad Pro 12.9” 2nd generation with Smart Keyboard

**Comment by carloscabanero** [comment]
Thanks a lot for the report and for testing!

Would you mind to elaborate more on what is exactly the problem? Unfortunately I don’t see it based on the video. Is it something with the cursor?

**Comment by kkk669** [comment]
I’m sorry for less information. To be accurate, the conversion candidate list should be displayed on the top of the screen in my video’s case because the cursor is there. But actually, the list is displayed on the bottom every time as you can see in my video.

**Comment by carloscabanero** [comment]
Phew! I thought it was going to be something harder to fix :)

We decided to do it down there becausje it makes it easier to select when you are typing with the external keyboard; if you have it down there then it is just a gesture to lift your finger and tap, but if we put it on top,  you have to raise your hand every time to select an option.

Do think it would be more comfortable up there following the cursor?

**Comment by kkk669** [comment]
As you said, the amount of movement of the hand is small in this implementation, and that makes it easier to select.

On the other hand, the amount of movement of the eyes can be huge especially when the cursor is on top of the screen. In that case, normally you look at the cursor position, but you have to look at the bottom only while inputting Japanese words, and nothing is presented on the cursor position until you select a one of conversion candidates.

I think Blink should show pre-converted Hiragana on the cursor position until conversion candidate is selected, and the conversion list should follow the cursor because of the above reasons.

In conclusion, I think the iOS’s normal behavior is more useful. Thanks.

**Comment by yury** [comment]
@kkk669 thank you for your feedback. We reimplemented input. So try it in next TF version

**Comment by kkk669** [comment]
I tested for a while. It looks nice. Thank you very much.
